src/tree.rs

```diff
@@ -714,7 +717,8 @@ impl<T> Tree<T> {
     /// # }
     /// ```
     ///
-    pub fn sort_children_by_key<B, F>(&mut self, node_id: &NodeId, mut f: F) -> Result<(), NodeIdError>
+    pub fn sort_children_by_key<B, F>(&mut self, node_id: &NodeId, mut f: F)
+                                      -> Result<(), NodeIdError>
         where B: Ord, F: FnMut(&Node<T>) -> B
     {
```

Comment:
I'm not sure how I feel about the formatting when we have a `return` that gets pushed to the next line and we also have a where clause (which is always pushed to the next line).

I feel like I would prefer it to be something like:
```
pub fn sort_children_by_key<B, F>(&mut self, node_id: &NodeId, mut f: F)
    -> Result<(), NodeIdError>
    where B: Ord, F: FnMut(&Node<T>) -> B
{
```
in that case, but I'm not really familiar enough with rustfmt to know what combinations of settings would produce that (if it's even possible) while leaving the rest of the behavior alone.

Any thoughts on this @Drakulix ?